### PR TITLE
Generate Orafin Feederfile

### LINKED ISSUE #728: check that the feeder file spec matches what we are doing
LIB File Format Position:
Please refer the format/fields position detail given below.
 
Header fields position:
source                          POSITION(1:3)   
internal_number        POSITION(4:13)
vendor_number          POSITION(16:21)
vendor_site_code       POSITION(22:36) 
invoice_number         POSITION(37:76) 
invoice_date               POSITION(77:84) 
invoice_amount         POSITION(85:99) 
invoice_type               POSITION(100:101) 
attribute1                   POSITION(102:123) 
due_date                    POSITION(124:131)
terms_name              POSITION(132:146)
attribute2                  POSITION(147:147)
 
Line fields position:
source                           POSITION(1:3)  
internal_number        POSITION(4:13) 
line_type                     POSITION(14:15)
amount                       POSITION(16:30)
tax_code                     POSITION(31:50)
project_number         POSITION(51:78)
task_number              POSITION(51:78)
award_number           POSITION(51:78)
expenditure                POSITION(51:78)
attribute5                   POSITION(51:78)
item_description       POSITION(79:115)
 
The last line fields position:
FEEDER_ID                         POSITION(1:3)  
TRAILER_NUMBER           POSITION(4:13) 
RECORD_TYPE                  POSITION(14:15) 
TRANSMISSION_DATE    POSITION(16:23) 
NO_OF_INVOICES            POSITION(24:29) 
BATCH_TOTAL_AMT       POSITION(30:44)

### LINKED ISSUE #729: newline missing in test feeder file on lines 7-9
Screen Shot 2023-08-04 at 9 14 32 AM

### LINKED ISSUE #730: make sure invoice lines total amounts equal invoice header total amount
In the test feeder file, an amount variance issue for 2 invoices was identified. Invoices 22-105280ZP1 10459, 15142 10592 header amount don’t match with lines amount total.

### PR DESCRIPTION
**Pull request by jermnelson**
Fixes #728
Fixes #729
Fixes #730

### CODE REVIEW

File: libsys_airflow/plugins/orafin/payments.py [hunk 1]
```diff
@@ -96,3 +108,11 @@ def models_converter():
     converter.register_structure_hook(datetime, _convert_ts_class)
     converter.register_unstructure_hook(datetime, datetime.isoformat)
     return converter
+
+
+def transfer_to_orafin(feeder_file_path: pathlib.Path, sftp_connection: str):
```
Comment (jermnelson) [inline libsys_airflow/plugins/orafin/payments.py:113]: Still need to figure out the best way to test the SFTP connection from either the stage or production Airflow server.

**Comment by shelleydoljack** [comment]
Closes #678

File: libsys_airflow/plugins/orafin/tasks.py [hunk 2]
```diff
@@ -71,7 +74,17 @@ def feeder_file_task(invoices: list):
     return converter.unstructure(feeder_file)
 
 
-# TODO: can un-ignore type checking here once this function is less of a stub
 @task
-def sftp_file_task(feeder_file: task):  # type: ignore
-    return "foo"
+def sftp_file_task(
```
Comment (jermnelson) [inline libsys_airflow/plugins/orafin/tasks.py:78]: Create a new task for generating the feeder file on disk and send the full path on Airflow to this task.

**Review by shelleydoljack — APPROVED** [review]
👍

**Pull request merged**